File: go/transaction_script/test/create_account_test.go
package test

import (
	"ex/transaction_script/proto"
	"ex/transaction_script/repository"
	"ex/transaction_script/service"
	"testing"

	scyna "github.com/scyna/core"
	scyna_test "github.com/scyna/core/testing"
)

func TestCreateShouldReturnSuccess(t *testing.T) {
	cleanup()
	scyna_test.EndpointTest(service.CREATE_ACCOUNT_URL).
		WithRequest(&proto.CreateAccountRequest{
			Email:    "a@gmail.com",
			Name:     "Nguyen Van A",
			Password: "1234565",
		}).
		ExpectSuccess().Run(t)

	scyna_test.EndpointTest(service.CREATE_ACCOUNT_URL).
		WithRequest(&proto.CreateAccountRequest{
			Email:    "a@gmail.com",
			Name:     "Nguyen Van A",
			Password: "1234565",
		}).
		ExpectError(repository.ACCOUNT_EXISTED).Run(t)

}

func TestCreateBadEmail(t *testing.T) {
	cleanup()
	scyna_test.EndpointTest(service.CREATE_ACCOUNT_URL).
		WithRequest(&proto.GetAccountResponse{
			Email: "a+gmail.com",
			Name:  "Nguyen Van A",
		}).
		ExpectError(scyna.REQUEST_INVALID).Run(t)
}

File: csharp/account_test/CreateAccountTest.cs
namespace ex.account.Test;

using NUnit.Framework;
using scyna;
using ex.account;

[TestFixture]
class CreateAccountTest
{
    [OneTimeSetUp]
    public void Setup()
    {
        Engine.Init("http://127.0.0.1:8081", "scyna_test", "123456");
        Endpoint.Register(Path.CREATE_ACCOUNT_URL, new CreateAccountService());
        Command.InitSingleWriter("ex_account");
    }

    [OneTimeTearDown]
    public void TearDown()
    {
        cleanup();
        Engine.Release();
    }

    [Test]
    public void TestCreateAccountShouldReturnSuccess()
    {
        EndpointTest.Create(Path.CREATE_ACCOUNT_URL)
                .WithRequest(new proto.CreateAccountRequest
                {
                    Email = "a@gmail.com",
                    Name = "Nguyen Van A",
                    Password = "12345678",
                })
                .MatchEvent(Path.ACCOUNT_CREATED_CHANNEL, new proto.AccountCreated
                {
                    Email = "a@gmail.com",
                    Name = "Nguyen Van A",
                })
                .ExpectSuccess()
                .Run();

        EndpointTest.Create(Path.CREATE_ACCOUNT_URL)
                .WithRequest(new proto.CreateAccountRequest
                {
                    Email = "a@gmail.com",
                    Name = "Nguyen Van A",
                    Password = "12345678",
                })
                .ExpectError(ex.account.Error.ACCOUNT_EXISTS)
                .Run();
    }

    [Test]
    public void TestCreateAccountWithBadEmail()
    {
        EndpointTest.Create(Path.CREATE_ACCOUNT_URL)
               .WithRequest(new proto.CreateAccountRequest
               {
                   Email = "a+gmail.com",
                   Name = "Nguyen Van A",
                   Password = "12345678",
               })
               .ExpectError(ex.account.Error.BAD_EMAIL)
               .Run();

        EndpointTest.Create(Path.CREATE_ACCOUNT_URL)
               .WithRequest(new proto.CreateAccountRequest
               {
                   Name = "Nguyen Van A",
                   Password = "12345678",
               })
               .ExpectError(ex.account.Error.BAD_EMAIL)
               .Run();

        EndpointTest.Create(Path.CREATE_ACCOUNT_URL)
               .WithRequest(new proto.CreateAccountRequest
               {
                   Email = "",
                   Name = "Nguyen Van A",
                   Password = "12345678",
               })
               .ExpectError(ex.account.Error.BAD_EMAIL)
               .Run();
    }

    private void cleanup()
    {
        var statement = Engine.DB.Session.Prepare("TRUNCATE ex_account.account").Bind();
        Engine.DB.Session.Execute(statement);
    }
}